sklearnex/preview/ensemble/forest.py

```diff
@@ -902,6 +922,11 @@ def _onedal_ready(self, X, y, sample_weight):
     def _onedal_cpu_supported(self, method_name, *data):
         if method_name == 'ensemble.RandomForestRegressor.fit':
             ready, X, y, sample_weight = self._onedal_ready(*data)
+            if self.splitter_mode == 'random':
+                warnings.warn("splitter_mode = 'random' supported on GPU device "
+                              "and requires OneDAL >= 2023.1.1. Using 'best' "

```

Comment:
Warning is not clear. I suggest something like next:
```suggestion
                warnings.warn("'random' splitter mode supports GPU devices only "
                              "and requires oneDAL version >= 2023.1.1. Using 'best' mode instead."
```

Reply:
Same for message in classifier